caas/kubernetes/provider/application/trust.go

```diff
@@ -12,23 +12,33 @@ import (
 	"github.com/juju/juju/caas/kubernetes/provider/resources"
 )
 
-var defaultApplicationNamespaceRules = []rbacv1.PolicyRule{
-	{
-		APIGroups: []string{""},
-		Resources: []string{"pods", "services"},
-		Verbs: []string{
-			"get",
-			"list",
-			"patch",
+func getDefaultApplicationNamespaceRules(namespace string) []rbacv1.PolicyRule {
```

Comment:
This change will disappear once rebase is done after https://github.com/juju/juju/pull/14699 merged.